app/src/selfRelease/java/org/radarcns/detail/UpdateAlarmReceiver.kt

```diff
@@ -0,0 +1,112 @@
+package org.radarcns.detail
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import org.radarcns.detail.UpdatesActivity.Companion.UPDATE_RELEASES_URL_KEY
+import org.radarcns.detail.UpdatesActivity.Companion.UPDATE_VERSION_NAME_KEY
+import org.radarcns.detail.UpdatesActivity.Companion.UPDATE_VERSION_URL_KEY
+import java.io.IOException
+
+class UpdateAlarmReceiver: BroadcastReceiver() {
+
+    private var updateNotificationConfig = true
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if(context == null) return
+
+        val url: String? = intent?.extras?.getString(UPDATE_RELEASES_URL_KEY)
+        if(url != null){
+            val request = okhttp3.Request.Builder().url(url).build()
+            val client = OkHttpClient()
```

Comment:
Better to use existing OkHttpClient thread queues:
```suggestion
            val client: OkHttpClient = RestClient.global()
                .httpClientBuilder()
                .build()
```
```suggestion
            val client = OkHttpClient()
```
```suggestion
            val client = OkHttpClient()
```